### PR TITLE
Namhyeong kim/support fp32 dest acc in moreh adam

### PR DESCRIPTION
**Pull request by namhyung-kim-moreh**
This PR supports fp32_dest_acc_en in moreh_adam.

### CODE REVIEW

File: tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_adam.py [hunk 1]
```diff
@@ -8,41 +8,71 @@
...
+            fp32_dest_acc_en=fp32_dest_acc_en,
+            packer_l1_acc=False,
+        )
+    else:
+        # Grayskull doesn't support fp32 but test passing a GS config is ok
+        compute_kernel_config = ttl.tensor.GrayskullComputeKernelConfig(
+            math_fidelity=ttl.tensor.MathFidelity.HiFi4,
+            math_approx_mode=True,
+        )
+    return compute_kernel_config
```
Comment (dongjin-na) [inline tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_adam.py:53]:
```suggestion
from tests.tt_eager.python_api_testing.unit_testing.misc.test_utils import (
    get_compute_kernel_options,
    compute_kernel_options,
    compute_kernel_ids,
)
```
You can use test_utils package.

File: tt_eager/tt_dnn/op_library/moreh_adam/kernels/moreh_adam.cpp [hunk 2]
```diff
@@ -72,306 +70,201 @@ void MAIN {
...
         cb_push_back(cb_tmp1, onetile);
-        REL();
+        tile_regs_release();
 
         // cb_tmp1 = 1 / (1 - cb_tmp1);
-        ACQ();
+        tile_regs_acquire();
         cb_wait_front(cb_tmp1, onetile);
         cb_reserve_back(cb_tmp1, onetile);
+        unpack_reconfig_data_format(cb_one, cb_tmp1);
```
Comment (dongjin-na) [inline tt_eager/tt_dnn/op_library/moreh_adam/kernels/moreh_adam.cpp:136]: How about adding #if defined (FP32_DEST_ACC_EN) macro guards around `unpack_reconfig_data_format()`?

File: tt_eager/tt_dnn/op_library/moreh_adam/moreh_adam_op.cpp [hunk 3]
```diff
@@ -75,77 +129,121 @@ operation::ProgramWithCallbacks MorehAdam::create_program(
...
-    const auto& max_exp_avg_sq = optional_input_tensors.at(0);
+    const auto& exp_avg_in = input_tensors.at(2);
+    const auto& exp_avg_sq_in = input_tensors.at(3);
+    const auto& max_exp_avg_sq_in = optional_input_tensors.at(0);
+
+    const auto& param_out = output_tensors.at(0);
+    const auto& exp_avg_out = output_tensors.at(1);
+    const auto& exp_avg_sq_out = output_tensors.at(2);
+    const auto& max_exp_avg_sq_out =
+        given_max_exp_avg_sq_in ? std::make_optional<Tensor>(output_tensors.at(3)) : std::nullopt;
```
Comment (hschoi4448) [inline tt_eager/tt_dnn/op_library/moreh_adam/moreh_adam_op.cpp:142]:
```suggestion
    const auto& max_exp_avg_sq_out =
        this->amsgrad ? std::make_optional<Tensor>(output_tensors.at(3)) : std::nullopt;
```
`max_exp_avg_sq_out` is not null, if `amsgrad` is true.

File: tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_adam.py (hunk 1 above)
Comment (namhyeong-kim) [reply tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_adam.py]: Done.

File: tt_eager/tt_dnn/op_library/moreh_adam/kernels/moreh_adam.cpp (hunk 2 above)
Comment (namhyeong-kim) [reply tt_eager/tt_dnn/op_library/moreh_adam/kernels/moreh_adam.cpp]:
```
#ifdef FP32_DEST_ACC_EN
#define WITH_FP32_DEST_ACC(x) x
#else
#define WITH_FP32_DEST_ACC(x)
#endif
```
I applied above macro.

File: tt_eager/tt_dnn/op_library/moreh_adam/moreh_adam_op.cpp (hunk 3 above)
Comment (namhyeong-kim) [reply tt_eager/tt_dnn/op_library/moreh_adam/moreh_adam_op.cpp]: Applied

**Review by dongjin-na — APPROVED** [review]
LGTM.

**Comment by sangwon-chae** [comment]
@tt-aho @mywoodstock @TT-BrianLiu Would you please review the PR? Thanks.

**Review by tt-aho — APPROVED** [review]

**Comment by hschoi4448** [comment]
Hello. I did a squash and merge, but I proceeded without changing the commit message.
In this case, how can I fix it?

@tt-rkim  @razorback3

**Comment by tt-rkim** [comment]
You cannot change it.

It is ok, but please be more careful next time. Thank you

**Pull request merged**